src/status_im/contexts/wallet/send/transaction_progress/view.cljs

```diff
@@ -25,6 +26,31 @@
     (some (fn [[_k v]] (= (:status v) :confirmed)) transaction-details)   :confirmed
     :else                                                                 nil))
 
+(defn show-save-address-bottom-sheet
+  ([] (show-save-address-bottom-sheet :blue))
+  ([_color]
+   (debounce/throttle-and-dispatch [:open-modal :screen/wallet.save-address] 1000)))
+
+(defn footer
+  [{:keys [color leave-page]}]
+  (let [save-address-visible? false]
+    [rn/view {:style style/footer}
+     (when save-address-visible?
+       [quo/button
+        {:type                :grey
+         :icon-left           :i/contact-book
+         :accessibility-label :save-address
+         :container-style     (style/footer-button save-address-visible?)
+         :on-press            (partial show-save-address-bottom-sheet color)}
```

Comment:
I'm not sure how `partial` memoizes something, but we might need to use `use-memo` if it's not.

Reply:
do we need to use partial here? why do we need to debounce the opening of the modal?

Reply:
Because if someone presses the button many times in succession, multiple modals will open up. This issue was raised by Pravesh in a call, and then he made the debounce handler.

Reply:
can we not abstract that away to the bottom sheet/modal?

Reply:
The `partial` is gone. But I'm not able to think of a way to abstract debounce. 

We are debouncing the event (from reaching the re-frame loop). So the debounce has to occur at the source of event creation.

Reply:
Update. This abstraction already exists. Removed manual throttle.